### PR TITLE
[MB-2220] Adds Message Center Support

### PR DESCRIPTION
**Pull request by gatlinhebert**

### CODE REVIEW

File: android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import com.urbanairship.reactnative.Event;
+
+/**
+ * Inbox updated event.
+ */
+public class InboxUpdatedEvent implements Event {
+
+    private static final String INBOX_UPDATED_EVENT = "com.urbanairship.inbox_updated";
+    private static final String MESSAGE_CENTER_UNREAD_COUNT = "messageCenterUnreadCount";
+    private static final String MESSAGE_CENTER_COUNT = "messageCenterCount";
```
Comment (rlepinski) [inline android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java]: messageCount

File: android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import com.facebook.react.bridge.WritableMap;
+import com.urbanairship.reactnative.Event;
+
+/**
+ * Inbox updated event.
+ */
+public class InboxUpdatedEvent implements Event {
+
+    private static final String INBOX_UPDATED_EVENT = "com.urbanairship.inbox_updated";
+    private static final String MESSAGE_CENTER_UNREAD_COUNT = "messageCenterUnreadCount";
```
Comment (rlepinski) [inline android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java]: messageUnreadCount

File: android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    private final int unreadCount;
+    private final int count;
+
+    /**
+     * Default constructor.
+     *
+     * @param unreadCount The number of unread messages in the message center.
+     * @param count The number of total messages in the message center.
+     */
+    public InboxUpdatedEvent(@NonNull int unreadCount, @NonNull int count) {
```
Comment (rlepinski) [inline android/src/main/java/com/urbanairship/reactnative/events/InboxUpdatedEvent.java]: primitive types are never null

File: ios/UARCTModule/UARCTEventEmitter.m [hunk 4]
```diff
@@ -154,6 +155,17 @@ - (void)notificationAuthorizedOptionsDidChange:(UANotificationOptions)options {
 }
 
 #pragma mark -
+#pragma mark Message Center
+
+- (void)inboxUpdated {
+    NSDictionary *body = @{ @"messageCenterUnreadCount": @([UAirship inbox].messageList.unreadCount),
```
Comment (rlepinski) [inline ios/UARCTModule/UARCTEventEmitter.m]: messageUnreadCount

File: ios/UARCTModule/UARCTEventEmitter.m [hunk 5]
```diff
@@ -154,6 +155,17 @@ - (void)notificationAuthorizedOptionsDidChange:(UANotificationOptions)options {
...
+                            @"messageCenterCount": @([UAirship inbox].messageList.messageCount)
```
Comment (rlepinski) [inline ios/UARCTModule/UARCTEventEmitter.m]: messageCount

File: js/UrbanAirship.js [hunk 6]
```diff
@@ -402,6 +405,89 @@ class UrbanAirship {
...
+  static setBadgeNumber(badgeNumber: number) {
+    UrbanAirshipModule.setBadgeNumber(badgeNumber);
+  }
+
+  /**
+   * Gets the current badge number for iOS.
+   *
+   * @return {Promise.<number>} A promise with the result.
+   */
+  static getBadgeNumber(): Promise<number> {
```
Comment (rlepinski) [inline js/UrbanAirship.js]: You need to rebase this branch on master and remove the commit for your other PR

**Comment by rlepinski** [comment]
Looks great, a few minor things to fix up

File: js/UrbanAirship.js [hunk 7]
```diff
@@ -438,28 +447,34 @@ class UrbanAirship {
...
-   * Retrieves the current inbox messages.
-   * 
+   * Retrieves the current inbox messages. Each message will have the following properties:
+   * "id": string - The messages ID. Needed to display, mark as read, or delete the message.
+   * "title": string - The message title.
+   * "sentDate": number - The message sent date in milliseconds.
+   * "listIconUrl": string, optional - The icon url for the message.
+   * "isRead": boolean - The unread/read status of the message.
+   * "extras": object - String to String map of any message extras.
+   *
```
Comment (gatlinhebert) [inline js/UrbanAirship.js]: "isDeleted": boolean - The deleted status of the message.

**Pull request merged**